seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java

```diff
@@ -67,9 +67,12 @@ public Object intercept(Object proxy, Method method, Object[] args, MethodProxy
                         result = get(convertDataId(rawDataId), args[1], (Long) args[2]);
                     }
                     if (null != result) {
+                        //If the return type is String,need to convert the object to string
+                        if (method.getReturnType().equals(String.class)) {
```

Comment:
What about the rest? Such as getBoolean 、 getShort、getInt、getLong？

Reply:
GetConfig converts the values of all configuration items to strings and returns them. Other methods, such as getShort, do not require such a transformation. And Boolean cannot be cast to String.